backend/capellacollab/settings/modelsources/t4c/injectables.py

```diff
@@ -12,7 +12,7 @@
 from capellacollab.settings.modelsources.t4c.models import DatabaseT4CInstance
 
 
-def load_instance(
+def get_existing_instance(
```

Comment:
This function either returns the `DatabaseT4CInstance` or raise the `HttpException`, so I think we can set the return type to simply `DatabaseT4CInstance`. This also has the benefit that we do not have to handle the `None` case at the places where we use this injectable. So for example, we can removed the `if not instance` in the `get_existing_instance_repository`.

Reply:
No, it returns None if t4c_instance_id is None.

Reply:
Yes, this is true, but I think we should not consider the cases where the id is `None`. So we use it as injectable on routes where we know the id exists (otherwise the route would not be called).

Reply:
Fixed.